File: src/main/java/com/company/payroll/mapper/LeaveMapper.java
package com.company.payroll.mapper;

import java.util.List;

import org.springframework.stereotype.Repository;

import com.company.payroll.model.Leave;

@Repository
public interface LeaveMapper {
    int deleteByPrimaryKey(Integer lId);

    int insert(Leave row);

    int insertSelective(Leave row);
    
    List<Leave> selectList();
    
    List<Leave> selectListByEId(int eid);
    
    Leave selectByPrimaryKey(Integer lId);

    int updateByPrimaryKeySelective(Leave row);

    int updateByPrimaryKey(Leave row);
}

File: kotlin-payroll/src/main/kotlin/com/company/payroll/mapper/LeaveMapper.kt
package com.company.payroll.mapper

import com.company.payroll.model.StaffLeave
import org.springframework.stereotype.Repository

@Repository
interface LeaveMapper {
  fun deleteByPrimaryKey(lId: Int): Int
  fun insert(row: StaffLeave): Int
  fun insertSelective(row: StaffLeave): Int
  fun selectList(): List<StaffLeave>
  fun selectListByEId(staffId: Int): List<StaffLeave>?
  fun selectByPrimaryKey(lId: Int): StaffLeave?
  fun updateByPrimaryKeySelective(row: StaffLeave): Int
  fun updateByPrimaryKey(row: StaffLeave): Int
}